### PR TITLE
modify HtmlPostProcessor.

### PR DESCRIPTION
**Pull request by vwxyzh**
@chenkennt @ansyral @hellosnow @vicancy @qinezh @superyyrrzz

### CODE REVIEW

**Comment by ansyral** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Build.Engine/PostProcessors/HtmlPostProcessContext.cs [hunk 1]
```diff
@@ -17,41 +16,41 @@ public class HtmlPostProcessContext
...
-                    var totalSrcFileSet = new HashSet<string>(host.SourceFileInfos.Select(s => s.SourceRelativePath));
-                    context.FileMapping = new OSPlatformSensitiveDictionary<string>(
-                        context.FileMapping
-                        .Where(p => totalSrcFileSet.Contains(p.Value))
-                        .ToDictionary(p => p.Key, p => p.Value));
-                    context.Bookmarks = new OSPlatformSensitiveDictionary<HashSet<string>>(
-                        context.Bookmarks
-                        .Where(p => context.FileMapping.ContainsKey(p.Key))
-                        .ToDictionary(p => p.Key, p => p.Value));
-                    return context;
```
Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.Engine/PostProcessors/HtmlPostProcessContext.cs]: the logic seems missing in handler?

File: src/Microsoft.DocAsCode.Build.Engine/PostProcessors/ValidateBookmark.cs [hunk 2]
```diff
@@ -4,35 +4,66 @@
...
+            {
+                throw new BuildCacheException("Registered bookmarks are not found in html post processor.");
+            }
+            var set = new HashSet<string>(
+                from sfi in context.PostProcessorHost.SourceFileInfos
+                where sfi.IsIncremental
+                select sfi.SourceRelativePath,
+                FilePathComparer.OSPlatformSensitiveStringComparer);
+            foreach (var pair in from f in fileMapping
+                                 where set.Contains(f.Key)
```
Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.Engine/PostProcessors/ValidateBookmark.cs]: >Key [](start = 54, length = 3)

Value?

File: src/Microsoft.DocAsCode.Build.Engine/PostProcessors/ValidateBookmark.cs [hunk 3]
```diff
@@ -4,35 +4,66 @@
...
+                                 select f)
+            {
+                _fileMapping[pair.Key] = pair.Value;
+            }
+            foreach (var pair in from b in registeredBookmarks
+                                 join f in _fileMapping on b.Key equals f.Value
```
Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.Engine/PostProcessors/ValidateBookmark.cs]: >Value [](start = 74, length = 5)

Key?

**Comment by ansyral** [comment]
👍

**Comment by hellosnow** [comment]
:shipit:

**Pull request merged**